### ISSUE TITLE
Not working when newrelic is installed

### ISSUE THREAD

**Issue by unustalha** [eKoopmans/html2pdf.js#220]
In my my react app i am trying to generate pdf by doing these
```
html2pdf(HTML_ELEMENT);
```

```
html2pdf().from(HTML_ELEMENT).set({
    margin: 0,
    filename: 'pricing-report.pdf',
    image: { type: 'jpeg', quality: 1 },
    html2canvas: {
        useCORS: true,
        scale: 2,
    },
    jsPDF: { unit: 'mm', format: 'a4', orientation: 'portrait' }
})
```

```
html2pdf(HTML_ELEMENT, {
    margin: 0,
    filename: 'pricing-report.pdf',
    image: { type: 'jpeg', quality: 1 },
    html2canvas: {
        useCORS: true,
        scale: 2,
    },
    jsPDF: { unit: 'mm', format: 'a4', orientation: 'portrait' }
})
```
but every time it return same error
```
Uncaught TypeError: Method Promise.prototype.then called on incompatible receiver #<Promise>
```

If i remove newrelic code it works fine. Looks like newrelic is overwriting `promise.prototype.then`

Is there any way to avoid this problem?

**Comment by tkimia** [comment]
I'm running into the same issue. Is there an update on this?

**Comment by kevinch** [comment]
Same error here in a React app but without any relic (btw, what is it?) code.

**Comment by kevinch** [comment]
Guys, i achieved a pdf doc from React with https://github.com/airarrazaval/html2pdf.

**Comment by unustalha** [comment]
@kevinch its application monitoring service. You can monitor web page load time, ajax request, js errors etc.
Thanks for the solution. i will check it.

**Comment by eKoopmans** [comment]
Hi, I'm planning to sandbox promises inside html2pdf.js to start avoiding these sorts of conflicts. Thanks for raising the issue!

**Comment by mzdr** [comment]
Hey guys,

we have the same problem. We are running a Vue application with webpack and (@babel/preset-env)[https://babeljs.io/docs/en/babel-preset-env] (using `useBuiltIns: 'usage'`) and keep getting the `Incompatible receiver, Promise required` error. If we turn transforms/polyfills (`useBuiltIns: false`) of it works. Even though our browserlist targets defines browsers that do not need Promise polyfilled, it still somehow seems to create conflicts with the Promise implementation/polyfill used by html2pdf.

**Comment by munsellj** [comment]
I'm experiencing a similar issue to what mzdr mentioned in a React app built with Create React App (CRA). After adding polyfills for IE 11 support PDF generation breaks in Firefox and Edge with a `TypeError: "Incompatible receiver, Promise required"` error (works fine still in Chrome, Safari and IE 11). If IE 11 polyfills aren't added to the project everything works fine.

Oddly enough another interesting thing is the issue only occurs in minified production builds (`npm build` from CRA). It isn't reproducible using the local development build and server (`npm start` from CRA).

**Comment by mzdr** [comment]
> Oddly enough another interesting thing is the issue only occurs in minified production builds[…]

Yes, we observed the same thing. We are using the "fork" @kevinch posted above and it works for now.

@eKoopmans I think the issue might be related to the fact that you create a Worker that inherits from the Promise.prototype and you may depend on a specific implementation of a polyfill? I am not 100% sure.

**Comment by unustalha** [comment]
It is still not working. I am also using fork posted by @kevinch. Thanks @kevinch

**Comment by DaddyWarbucks** [comment]
I am also still experiencing this problem on version `0.9.1`.  I receive `TypeError: Incompatible receiver, Promise required` on Firefox and Edge. I am also using some polyfills for older browsers.

**Comment by VanPaitin** [comment]
This issue is not related to new relic. I turned off new relic but I am still getting the same error

**Comment by eKoopmans** [comment]
I'm sorry, the issue was fixed but never released in a new version. It should be resolved in v0.9.2 (released today).
